File: src/main/java/com/refinedmods/refinedstorage/screen/factory/CrafterManagerScreenFactory.java
package com.refinedmods.refinedstorage.screen.factory;

import com.refinedmods.refinedstorage.container.CrafterManagerContainerMenu;
import com.refinedmods.refinedstorage.screen.CrafterManagerScreen;
import net.minecraft.client.gui.screens.MenuScreens;
import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.player.Inventory;

public class CrafterManagerScreenFactory implements MenuScreens.ScreenConstructor<CrafterManagerContainerMenu, CrafterManagerScreen> {
    @Override
    public CrafterManagerScreen create(CrafterManagerContainerMenu container, Inventory playerInventory, Component title) {
        CrafterManagerScreen screen = new CrafterManagerScreen(container, playerInventory, title);

        container.setScreenInfoProvider(screen);

        return screen;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/factory/CrafterManagerScreenFactory.kt
package com.refinedmods.refinedstorage.screen.factory

import com.refinedmods.refinedstorage.container.CrafterManagerContainer
import com.refinedmods.refinedstorage.screen.CrafterManagerScreen
import net.minecraft.client.gui.ScreenManager
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.util.text.Text

class CrafterManagerScreenFactory : ScreenManager.IScreenFactory<CrafterManagerContainer?, CrafterManagerScreen?> {
    fun create(container: CrafterManagerContainer, playerInventory: PlayerInventory?, title: Text?): CrafterManagerScreen {
        val screen = CrafterManagerScreen(container, playerInventory, title)
        container.setScreenInfoProvider(screen)
        return screen
    }
}